### DISCUSSION TITLE
I am a beginner，how can I draw some lines or simple shapes (such as rectangles) to the scene?

### DISCUSSION THREAD

**Discussion by aliule2020** [stride3d/stride#1357] [Q&A] [answered]
I am a beginner，how can I draw some lines or simple shapes (such as rectangles) to the scene? Can you give a simple program?

**Comment by VaclavElias** [comment]
I am trying to figure out the same, here is my example 

![image](https://user-images.githubusercontent.com/4528464/159271460-fc5c0397-ff51-4c23-a2ed-0ad674287106.png)

https://github.com/VaclavElias/Stride3DTutorials/blob/f0b7f186c6a9de8f0f6e420051fc40773867873e/Other/Minimal/Minimal/Program.cs#L9-L72

I added there some spheres so I can understand how it works. You would need to update this code because my code works only with the master branch + code-only approach. I am preparing a full example which will be available later here https://github.com/VaclavElias/stride-code-only.

**Reply by aliule2020** [reply]
Thank you for your answer, it is good and powerfull. But I think it will be better if Strde Engine provide a higher abstract layer to finish some simple task, rather than just providing OpenGL-like low level functions(though I think it may be more efficient), after all we want to use a simple easy-to-use powerfull C# game engine. Of course, it is just my opinion.

**Reply by tebjan** [reply]
hey, there is a much easier solution, as you describe. have a look here: https://github.com/stride3d/stride/discussions/1041

**Comment by ykafia** [comment, marked as answer]
As @tebjan mentionned, there is a solution in #1041
